controllers/cloud.redhat.com/mutant.go

```diff
@@ -0,0 +1,126 @@
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	crd "cloud.redhat.com/clowder/v2/apis/cloud.redhat.com/v1alpha1"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type mutantAnnotatorApp struct {
+	Client   client.Client
+	Recorder record.EventRecorder
+	decoder  *admission.Decoder
+}
+
+func (a *mutantAnnotatorApp) Handle(ctx context.Context, req admission.Request) admission.Response {
+	app := &crd.ClowdApp{}
+	err := a.decoder.Decode(req, app)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	if app.Spec.Pods != nil {
+		// TODO find out why events are not submitted
```

Comment:
Is this a stretch or a TODO before merging?

Reply:
Stretch - it's not _super_ important but would be nice